### ISSUE TITLE
Oracle mysql compatibility consultion

### ISSUE THREAD

**Issue by Jacklli** [presslabs/mysql-operator#413]
dear sir,
may i ask that whether this mysql-operator is compatible with official Oracle mysql?
we use official Oracle mysql? it's important for us.
thanks very much.

**Comment by calind** [comment]
It should work with MySQL 5.7 out of the box but we don't test for it right now. If you want to add support for it we are happy to review and integrate PRs regarding this issue.

**Comment by Jacklli** [comment]
could you please provide us an docker image with oracle mysql 5.7，or give us an introduction about  how to make docker image that can run this operator？

**Comment by AMecea** [comment]
You can try using those docker images, those are from Oracle.

https://hub.docker.com/r/mysql/mysql-server/

Please let us know if you encounter any issues with those images.

**Comment by Jacklli** [comment]
thanks a lot, i will have a try. By the way, may i ask that where does the third party tools located? such as orchestrator. is it in the mysql image or some where else?

**Comment by AMecea** [comment]
The orchestrator is deployed along with the operator, in the same pod. And some other tooling is deployed with the server, for example, Percona Toolkit is used for replication monitoring, and query limit enforcement, and metrics exporter.

Maybe this [presentation](https://docs.google.com/presentation/d/1UXtL18ZC-IXUp9avjEzD4PIRUmGx3UGQyy6ma6BIpq8/edit#slide=id.g5102507ae4_0_731) from KubeCon 2019 will be helpful (see slides 14 and 16).

**Comment by Jacklli** [comment]
thanks a lot

**Comment by AMecea** [comment]
The tools are deployed in the sidecar image (a different image than the MySQL image), so you can use another MySQL image without affecting the tooling.

**Comment by Jacklli** [comment]
got it，thanks very much

**Comment by Jacklli** [comment]
@AMecea hello, may i ask that how could i change the mysql image? i tried, but there's no entry that can configure the image, either mysql-cluster yaml or mysql-operator yaml.

**Comment by AMecea** [comment]
This image can be changed per cluster. Here is an [example](https://github.com/presslabs/mysql-operator/blob/1169dc18e866541319ffad413e1c38c0401bcba3/examples/example-cluster.yaml#L11).

Hope this helps!

**Comment by Jacklli** [comment]
@AMecea if i want to change Percona server to Oracle official mysql, what should i configure?
i checked the source code, it seems there's only percona server images can be chose. should i change the source code first before i configure the yaml file?

var (
        // MySQLDefaultVersion is the version for mysql that should be used
        MySQLDefaultVersion = semver.MustParse("5.7.26")
        // MySQLTagsToSemVer maps simple version to semver versions
        MySQLTagsToSemVer = map[string]string{
                "5.7": "5.7.26",
        }
        // MysqlImageVersions is a map of supported mysql version and their image
        MysqlImageVersions = map[string]string{
                // Percona:5.7.26 CentOS based image
                "5.7.26": "percona@sha256:713c1817615b333b17d0fbd252b0ccc53c48a665d4cfcb42178167435a957322",
                // Percona:5.7.24 CentOS based image
                "5.7.24": "percona@sha256:b3b7fb177b416563c46fe012298e042ec1607cc0539ce6014146380b0d27b08c",
        }
)

**Comment by AMecea** [comment]
No, you can choose the Oracle image of MySQL 5.7 but some features will not work, it's about [the initialization of MySQL in a separate init container,](https://github.com/presslabs/mysql-operator/issues/229#issuecomment-465202040) this is not a critical fix, maybe it won't affect you.

**Comment by Jacklli** [comment]
@AMecea thanks

**Comment by Jacklli** [comment]
@AMecea could you please have a look?

after change the mysql image from percona to oracle mysql image (5.7.27), there's an error occured:
when connecting to mysql through 'my-cluster-mysql-master ', ssl error returned, while through ' my-cluster-mysql-replicas', it succeeded.

[root@k8smaster examples]# kubectl get svc
NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
kubernetes                  ClusterIP   10.96.0.1        <none>        443/TCP             3h21m
my-cluster-mysql            ClusterIP   10.102.52.180    <none>        3306/TCP            2m26s
my-cluster-mysql-master     ClusterIP   10.100.150.191   <none>        3306/TCP,8080/TCP   2m26s
my-cluster-mysql-replicas   ClusterIP   10.96.129.27     <none>        3306/TCP,8080/TCP   2m26s
mysql                       ClusterIP   None             <none>        3306/TCP,9125/TCP   168m
mysql-operator              ClusterIP   10.111.201.234   <none>        80/TCP              48m
mysql-operator-0-svc        ClusterIP   10.101.128.112   <none>        80/TCP,10008/TCP    48m
[root@k8smaster examples]# mysql -h 10.100.150.191 -uroot -pmypass
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 2026 (HY000): SSL connection error: error:1425F102:SSL routines:ssl_choose_client_version:unsupported protocol
[root@k8smaster examples]# mysql -h 10.96.129.27 -uroot -pmypass
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 79
Server version: 5.7.27-log MySQL Community Server (GPL)

Copyright (c) 2000, 2019, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>
